### DISCUSSION TITLE
Raspberry Pi Zero with Sonoff Stick: z2m on RPi or ser2net ?

### DISCUSSION THREAD

**Discussion by eigenphase** [Koenkk/zigbee2mqtt#17889] [General]
I am running a Raspberry Pi Zero W with rtl433 and Zigbee (via Sonoff stick). It's connected via PoE HAT (no Wifi).

Home Assistant runs separately with the mosquitto add-on.

Would you

1. Run zigbee2mqtt directly on the Pi Zero and let the results publish over the network via MQTT, or 
2. Run ser2net on the Pi Zero and run zigbee2mqtt as add-on in Home Assistant (and let it connect to the networked serial adapter)

Why / Why not?

I am currently doing (1) which I feel is the most elegant. On the other hand, the armv61 port of node.js is only unofficial and z2m is fairly heavy weight. I don't want to overload my Pi and it sucks already a few watts (with the PoE hat, the SDR stick and the Sonoff stick). I am wondering if (2) would be considered a better option.

**Comment by deviantintegral** [comment]
I ran option 1 on wifi, which I think was best. While it was "heavier", for a 10 device network z2m was pretty much idle the whole time. I ran into trouble with the Pi seriously overheating and crashing (it melted the case I had it in!) but I think that is probably a hardware issue and no fault of z2m. I still need to confirm that though.

The main reason I did it that way was that mqtt should be pretty good at handling disconnects and reconnects. I had read (but not confirmed) that was an issue with ser2net style connections.

The one downside was that there wasn't enough ram to build z2m on later nodejs versions. I ended up switching to docker containers to pull it all down. I also configured docker to send logs to journald / systemd, and then set the journald logs to ram-only to reduce SSD wear.

With a wired network, it's less clear-cut. I would probably try it first with only 1-2 zigbee devices, and make sure it reconnects nicely both when the Pi zero reboots and if it keeps power but loses the network connection.

If things are working fine with z2m on the pi, it's probably not worth changing.
